projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,287 @@
+
+      this.arrivalTime = arrivalTime;
+      setStop(stopTime);
+    }
+
+    private void setStop(StopTime stopTime) throws IOException {
+
+      Stop stopAround;
+      try (FileInputStream stopInputStream =
+          new FileInputStream(new File("projects/opentrain/gtfs_pipeline/stops.protobin"))) {
+
+        // From all stops class pick the chosen stop from stopsTime *to use longitude/latitude
+        while (true) {
+
+          stopAround = Stop.parseDelimitedFrom(stopInputStream);
+
+          if (stopAround == null) {
+            break;
+          }
+          if (stopAround.getStopId() == stopTime.getStopId()) {
+            stop = stopAround;
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  // arrivalTime To Location by stops.latitude,stops.longitude make matrix of colors
+  private void arrivalTimeToLocation() {
+
+    double maxLongitude = 0;
+    double maxLatitude = 0;
+
+    for (StopsAround stp : stopsAroundList) {
+      if (stp.stop.getStopLat() > maxLatitude) {
+        maxLatitude = (stp.stop.getStopLat());
+      }
+
+      if (stp.stop.getStopLon() > maxLongitude) {
+        maxLongitude = stp.stop.getStopLon();
+      }
+    }
+
+    // Location matrix *100 to make difference between points
+    heatMap = new int[(int) (maxLatitude * 100) + 1][(int) (maxLongitude * 100) + 1];
+
+    for (StopsAround stp : stopsAroundList) {
+
+      // Arrival time of each stop from the chosen stop by location
+      // *100 to make difference between location
+      stp.i = (int) (stp.stop.getStopLat() * 100);
+      stp.j = (int) (stp.stop.getStopLon() * 100);
+      heatMap[stp.i][stp.j] = (int) (stp.arrivalTime);
+    }
+
+    // Distance of other points to the nearest stop
+
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+
+        if (heatMap[i][j] == 0) // not a stop
+        {
+          StopsAround firstStop = stopsAroundList.get(0);
+          double ac = Math.abs(0 - firstStop.i); // y distance
+          double cb = Math.abs(0 - firstStop.j); // x distance
+          double distance = Math.hypot(ac, cb);
+          double stopTimeArrival = firstStop.arrivalTime;
+
+          // Find nearest stop
+          for (StopsAround stp : stopsAroundList) {
+
+            ac = Math.abs(i - stp.i); // y distance
+            cb = Math.abs(j - stp.j); // x distance
+
+            if (distance > Math.hypot(ac, cb)) {
+              distance = Math.hypot(ac, cb);
+              stopTimeArrival = stp.arrivalTime;
+            }
+          }
+          // *60 minutes to seconds
+          heatMap[i][j] = (int) ((distance * 60) + stopTimeArrival);
+        }
+        if (heatMap[i][j] == 0) {
+          System.out.println("Check heatMap assignment");
+        }
+      }
+    }
+  }
+
+  public HeatMap(int stopId, String time) throws IOException { // Time = HH:MM:ss
+
+    StopTime stopTime;
+    Trip trip;
+
+    try (FileInputStream tripsInputStream =
```

Comment:
I think you could simplify the code if you load `trips.protobin` and `stopTime.protobin` into lists in memory.
They are small enough to fit. and it will also reduce I/O.